solutions/0000-0099/0036-valid-sudoku-easy.md

```diff
@@ -1,5 +1,168 @@
+
+**Note:**
+* A Sudoku board (partially filled) could be valid but is not necessarily solvable.
+* Only the filled cells need to be validated according to the mentioned rules.
+
+**Example 1:**
+
+```
+5 3 .  . 7 .  . . .  
+6 . .  1 9 5  . . .  
+. 9 8  . . .  . 6 .  
+
+8 . .  . 6 .  . . 3  
+4 . .  8 . 3  . . 1  
+7 . .  . 2 .  . . 6  
+
+. 6 .  . . .  2 8 .  
+. . .  4 1 9  . . 5  
+. . .  . 8 .  . 7 9 
+```
+
+```
+Input: board = 
+[["5","3",".",".","7",".",".",".","."]
+,["6",".",".","1","9","5",".",".","."]
+,[".","9","8",".",".",".",".","6","."]
+,["8",".",".",".","6",".",".",".","3"]
+,["4",".",".","8",".","3",".",".","1"]
+,["7",".",".",".","2",".",".",".","6"]
+,[".","6",".",".",".",".","2","8","."]
+,[".",".",".","4","1","9",".",".","5"]
+,[".",".",".",".","8",".",".","7","9"]]
+Output: true
+```
+
+**Example 2:**
+
+```
+Input: board = 
+[["8","3",".",".","7",".",".",".","."]
+,["6",".",".","1","9","5",".",".","."]
+,[".","9","8",".",".",".",".","6","."]
+,["8",".",".",".","6",".",".",".","3"]
+,["4",".",".","8",".","3",".",".","1"]
+,["7",".",".",".","2",".",".",".","6"]
+,[".","6",".",".",".",".","2","8","."]
+,[".",".",".","4","1","9",".",".","5"]
+,[".",".",".",".","8",".",".","7","9"]]
+Output: false
+Explanation: Same as Example 1, except with the 5 in the top left corner being modified to 8. Since there are two 8's in the top left 3x3 sub-box, it is invalid.
+```
+
+**Constraints:**
+
+* `board.length == 9`
+* `board[i].length == 9`
+* `board[i][j]` is a digit `1-9` or `'.'`.
+
+## Approach 1: HashSet
+
+Every value in the row or column or `3 x 3` block, the value is unique, With the help of this simple logic, add the Numbers in the HashSet, Add `num` value for `m`th row and for `n`th column and the same for `3 x 3` matrix. 
+
+If the same value repeat again for the same row or column or `3 x 3` block, then the given sudoku is not valid, otherwise is valid.
+
+Time Complexity: O(m * n), where m - # of rows, n - # of cols
+Space complexity: O(1), since the values are fixed for `9 x 9` board
+
+<Tabs>
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@vigneshshiv"/>
+
+```java
+class Solution {
+    public boolean isValidSudoku(char[][] board) {
+        Set<String> seen = new HashSet<>();
+        for (int r = 0; r < 9; r++) {
+            for (int c = 0; c < 9; c++) {
+                int num = board[r][c];
+                if (num != '.') {
+                    if (!seen.add(num + " in row " + r) || !seen.add(num + " in column " + c)
+                            || !seen.add(num + " in block " + (r / 3) + "-" + (c / 3))) {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Standard
+
+Iterate over rows and columns and `3 x 3` block to check if the number is repeats. 
+
+Time complexity: O(9^n^2)
```

Comment:
9^n^2???

Reply:
Time complexity mentioned for this problem is wrong, O(9^n^2) is actually for Sudoku Solver problem. 

Correct Time complexity for this problem is O(n^2)